test/built-ins/Array/prototype/findIndex/predicate-call-parameters.js

```diff
@@ -0,0 +1,59 @@
+// Copyright (C) 2015 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 22.1.3.9
+description: >
+  Predicate called as F.call( thisArg, kValue, k, O ) for each array entry.
+info: >
+  22.1.3.9 Array.prototype.findIndex ( predicate[ , thisArg ] )
+
+  ...
+  6. If thisArg was supplied, let T be thisArg; else let T be undefined.
+  7. Let k be 0.
+  8. Repeat, while k < len
+    ...
+    d. Let testResult be ToBoolean(Call(predicate, T, «kValue, k, O»)).
+  ...
+---*/
+
+var arr = ['Mike', 'Rick', 'Leo'];
+
+var results = [];
+
+arr.findIndex(function(kValue, k, O) {
+  results.push({
+    kValue: kValue,
+    k: k,
+    O: O
+  });
+});
+
+var result = results[0];
+assert.sameValue(result.kValue, 'Mike');
+assert.sameValue(result.k, 0);
+assert.sameValue(result.O, arr);
+assert.sameValue(result.thisArg, undefined);
+
+result = results[1];
+assert.sameValue(result.kValue, 'Rick');
+assert.sameValue(result.k, 1);
+assert.sameValue(result.O, arr);
+assert.sameValue(result.thisArg, undefined);
+
+result = results[2];
+assert.sameValue(result.kValue, 'Leo');
+assert.sameValue(result.k, 2);
+assert.sameValue(result.O, arr);
+assert.sameValue(result.thisArg, undefined);
+
+assert.sameValue(results.length, 3);
```

Comment:
Better to test `results.length` first (that way, if the number is less than  expected, a descriptive assertion error will be thrown instead of a TypeError for `results[2].kValue`)